aidial_assistant/tools_chain/tools_chain.py

```diff
@@ -37,41 +104,46 @@ def _publish_command(
     args_callback.on_args_end()
 
 
+CommandTool = Tuple[CommandConstructor, ChatCompletionToolParam]
+CommandToolDict = dict[str, CommandTool]
+
+
 class ToolCallsCallback(ExtraResultsCallback):
     def __init__(self):
-        self.tool_calls: list[ToolCall] = []
+        self.tool_calls: list[ChatCompletionMessageToolCallParam] = []
 
-    def on_tool_calls(self, tool_calls: list[ToolCall]):
+    def on_tool_calls(
+        self, tool_calls: list[ChatCompletionMessageToolCallParam]
+    ):
         self.tool_calls = tool_calls
 
 
 class ToolsChain:
-    def __init__(
-        self,
-        model: ModelClient,
-        tools: list[Tool],
-        command_dict: CommandDict,
-    ):
+    def __init__(self, model: ModelClient, command_tool_dict: CommandToolDict):
```

Comment:
I think `commands` will work just fine. No need to repeat the whole type in the variable name.

Reply:
Renamed.